src/Features/Core/Portable/RemoveUnusedMembers/AbstractAvoidUnusedMembersCodeFixProvider.cs

```diff
@@ -0,0 +1,116 @@
+            Document document, SyntaxEditor editor, ImmutableArray<Diagnostic> diagnostics,
+            CancellationToken cancellationToken)
+        {
+            var declarators = new HashSet<SyntaxNode>();
+            var fieldDeclarators = new HashSet<TFieldDeclarationSyntax>();
+
+            // Compute declarators to remove, and also track common field declarators.
+            foreach (var diagnostic in diagnostics)
+            {
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var diagnosticSpan = diagnostic.Location.SourceSpan;
+                var symbolName = diagnostic.Properties[AvoidUnusedMembersDiagnosticAnalyzer.UnunsedMemberNameProperty];
+                var symbolKind = diagnostic.Properties[AvoidUnusedMembersDiagnosticAnalyzer.UnunsedMemberKindProperty];
+
+                var node = GetTopmostSyntaxNodeForSymbolDeclaration(root.FindNode(diagnosticSpan),
+                    isSymbolDeclarationNode: n => n != null && semanticModel.GetDeclaredSymbol(n, cancellationToken)?.Name == symbolName);
+
+                declarators.Add(node);
+                if (symbolKind == nameof(SymbolKind.Field))
+                {
+                    var fieldDeclarator = node.FirstAncestorOrSelf<TFieldDeclarationSyntax>();
+                    fieldDeclarators.Add(fieldDeclarator);
+                }
+            }
+
+            if (fieldDeclarators.Count > 0)
+            {
+                AdjustDeclarators(fieldDeclarators, declarators);
+            }
+
+            foreach (var declarator in declarators)
+            {
+                editor.RemoveNode(declarator);
+            }
+        }
+
+        protected virtual SyntaxNode GetTopmostSyntaxNodeForSymbolDeclaration(SyntaxNode syntaxNode, Func<SyntaxNode, bool> isSymbolDeclarationNode)
+        {
+            return syntaxNode.FirstAncestorOrSelf(isSymbolDeclarationNode);
+        }
+
+        protected void AdjustChildDeclarators(SyntaxNode parentDeclaration, IEnumerable<SyntaxNode> childDeclarators, HashSet<SyntaxNode> declarators)
+        {
+            Debug.Assert(!declarators.Contains(parentDeclaration));
+
+            var declaratorsContainsAllChildren = true;
+            foreach (var childDeclarator in childDeclarators)
+            {
+                if (!declarators.Contains(childDeclarator))
+                {
+                    declaratorsContainsAllChildren = false;
+                    break;
+                }
+            }
+
+            if (declaratorsContainsAllChildren)
+            {
+                // Remove the entire parent declaration instead of individual child declarators within it.
+                declarators.Add(parentDeclaration);
+                declarators.RemoveAll(childDeclarators);
+            }
+        }
```

Comment:
TODO(cyrusn).  Have not reviewed this yet.  waiting to get explanation/comments as to what this all for.